apps/web/server/routers/viewer.tsx

```diff
@@ -940,6 +942,244 @@ const loggedInViewerRouter = createProtectedRouter()
+      id: z.number(),
+    }),
+    async resolve({ input, ctx }) {
+      const { id } = input;
+
+      const credential = await prisma.credential.findFirst({
+        where: {
+          id: id,
+        },
+        include: {
+          app: {
+            select: {
+              slug: true,
+              categories: true,
+            },
+          },
+        },
+      });
+
+      if (!credential) {
+        throw new TRPCError({ code: "INTERNAL_SERVER_ERROR" });
+      }
+
+      if (credential.userId !== ctx.user.id) {
+        throw new TRPCError({ code: "UNAUTHORIZED" });
+      }
+
+      const eventTypes = await prisma.eventType.findMany({
+        where: {
+          users: {
+            some: {
+              id: ctx.user.id,
+            },
+          },
+        },
+        select: {
+          id: true,
+          locations: true,
+          destinationCalendar: true,
+          price: true,
+        },
+      });
+
+      // If it's a video, replace the location with Cal video
+      if (credential.app?.categories.includes(AppCategories.video)) {
+        // Find the user's event types
+
+        // Look for integration name from app slug
+        const integrationQuery =
+          credential.app?.slug === "msteams" ? "office365_video" : credential.app?.slug.split("-")[0];
+
+        // Check if the event type uses the deleted integration
+        for (const eventType of eventTypes) {
+          if (isPrismaArray(eventType.locations)) {
+            // To avoid type errors, need to stringify and parse JSON to use array methods
+            const locationsSchema = z.array(z.object({ type: z.string() }));
+            const locations = locationsSchema.parse(eventType.locations);
+
+            const updatedLocations = locations.map((location: { type: string }) => {
+              if (location.type.includes(integrationQuery)) {
+                return { type: "integrations:daily" };
+              }
+              return location;
+            });
+
+            await prisma.eventType.update({
+              where: {
+                id: eventType.id,
+              },
+              data: {
+                locations: updatedLocations,
+              },
+            });
+          }
+        }
+      }
+
+      // If it's a calendar, remove the destination claendar from the event type
+      if (credential.app?.categories.includes(AppCategories.calendar)) {
+        for (const eventType of eventTypes) {
+          if (eventType.destinationCalendar?.integration === credential.type) {
+            await prisma.destinationCalendar.delete({
+              where: {
+                id: eventType.destinationCalendar.id,
+              },
+            });
+          }
+        }
+      }
+
+      // If it's a payment, hide the event type and set the price to 0. Also cancel all pending bookings
+      if (credential.app?.categories.includes(AppCategories.payment)) {
+        for (const eventType of eventTypes) {
```

Comment:
Third time looping eventTypes

Reply:
Why not instead loop one time and run every condition inside?

Reply:
Great idea, implemented